### PR TITLE
add out prefix for mm jobs

### PR DESCRIPTION
**Pull request by JackTemaki**

### CODE REVIEW

**Review by christophmluscher — APPROVED** [review]

**Review by mmz33 — APPROVED** [review]

File: meta/system.py [hunk 1]
```diff
@@ -388,7 +388,7 @@ def linear_alignment(self, name, corpus, flow, prefix="", **kwargs):
             self.crp[corpus], self.feature_flows[corpus][flow], **kwargs
         )
         j.add_alias(prefix + name)
-        self.mixtures[corpus][name] = j.mixtures
+        self.mixtures[corpus][name] = j.out_mixtures
         if hasattr(j, "alignment_path") and hasattr(j, "alignment_bundle"):
```
Comment (michelwi) [inline meta/system.py:392]: should this also be `out_alignment_path` and `out_alignment_bundle`?

Ah you did not modify the `LinearAlignmentJob` intentionally?

File: meta/warping_sequence.py [hunk 2]
```diff
@@ -48,9 +48,9 @@ def __init__(
             j = vtln.EstimateWarpingMixturesJob(**args)
             self.all_jobs.append(j)
             self.all_logs.append(j.log_file)
```
Comment (michelwi) [inline meta/warping_sequence.py:50]: `out_log_file`?

File: mm/alignment.py [hunk 3]
```diff
@@ -55,27 +55,32 @@ def __init__(
...
+            "alignment.cache.bundle", cached=True
         )
-        self.alignment_bundle = self.output_path("alignment.cache.bundle", cached=True)
         if self.word_boundaries:
-            self.single_word_boundary_caches = dict(
+            self.out_single_word_boundary_caches = dict(
                 (i, self.output_path("word_boundary.cache.%d" % i, cached=True))
                 for i in range(1, self.concurrent + 1)
             )
             self.word_boundary_path = util.MultiOutputPath(
```
Comment (michelwi) [inline mm/alignment.py:77]: `out_word_boundary_path`?

File: mm/mixtures.py [hunk 4]
```diff
@@ -60,14 +60,14 @@ def inc_merge_count(e):
             util.partition_into_tree(mixtures_to_combine, combine_per_step),
         )
 
-        self.merge_log_file = self.log_file_output_path("merge", crp, merge_count)
-        self.mixtures = self.output_path("am.mix", cached=True)
+        self.out_merge_log_file = self.log_file_output_path("merge", crp, merge_count)
+        self.out_mixtures = self.output_path("am.mix", cached=True)
 
-        self.merge_rqmt = {"time": max(merge_count / 25, 0.5), "cpu": 1, "mem": 1}
```
Comment (michelwi) [inline mm/mixtures.py:66]: there are several Jobs that inherit from MergeMixturesJob (e.g. LinearAlignmentJob, EstimateMixturesJob), which have different rqmt for their working step and only do the merging after they are done. Their rqmt would overwrite this merging related rqmt if it were not called differently.

File: meta/mm_sequence.py [hunk 5]
```diff
@@ -154,7 +154,7 @@ def update_rqmt(rqmt, extra):
                 self.all_jobs.append(job)
                 self.all_logs.append(job.log_file)
```
Comment (michelwi) [inline meta/mm_sequence.py:155]: would you not also add an out_ prefix to `EstimateMixturesJob.log_file`?

**Review by michelwi — CHANGES_REQUESTED** [review]
I found several out_ prefixes still missing, was this intentional?

You should not rename the merge_rqmt without discussion.

File: meta/system.py (hunk 1 above)
Comment (JackTemaki) [reply meta/system.py:392]: Oh, this `hasattr` stuff is dangerous...

**Comment by JackTemaki** [comment]
> I found several out_ prefixes still missing, was this intentional?
> 
> You should not rename the merge_rqmt without discussion.

@michelwi No this was not intentional, I hope I got it all now...

**Review by curufinwe — APPROVED** [review]

**Review by michelwi — APPROVED** [review]

**Pull request merged**